src/Agent/NewRelic/Agent/Core/DataTransport/DataStreamingService.cs

```diff
@@ -382,6 +387,17 @@ public bool ReadAndValidateConfiguration()
             return false;
         }
 
+        private void CheckForLegacyProxyAndDisplayWarning()
+        {
+            var grpcProxyValue = _environment.GetEnvironmentVariable("grpc_proxy");
+            var httpsProxyValue = _environment.GetEnvironmentVariable("https_proxy");
+
+            if (!string.IsNullOrWhiteSpace(grpcProxyValue) && string.IsNullOrWhiteSpace(httpsProxyValue))
+            {
+                LogMessage(LogLevel.Warn, "The 'grpc_proxy' environment variable was detected and the 'https_proxy' environment variable was not detected. The 'grpc_proxy' environment variable is not supported by all gRPC libraries. If a proxy is needed consider using the 'https_proxy' environment variable.");
```

Comment:
This is fine, but it probably doesn't need to be this verbose? Maybe consider something like `The grpc_proxy environment variable is deprecated/no longer supported. Please use http_proxy instead.`?